set-galasa-boot-version.sh

```diff
@@ -0,0 +1,159 @@
+underline() { printf "${underline}${bold}%s${reset}\n" "$@" ;}
+h1() { printf "\n${underline}${bold}${blue}%s${reset}\n" "$@" ;}
+h2() { printf "\n${underline}${bold}${white}%s${reset}\n" "$@" ;}
+debug() { printf "${white}%s${reset}\n" "$@" ;}
+info() { printf "${white}➜ %s${reset}\n" "$@" ;}
+success() { printf "${green}✔ %s${reset}\n" "$@" ;}
+error() { printf "${red}✖ %s${reset}\n" "$@" ;}
+warn() { printf "${tan}➜ %s${reset}\n" "$@" ;}
+bold() { printf "${bold}%s${reset}\n" "$@" ;}
+note() { printf "\n${underline}${bold}${blue}Note:${reset} ${blue}%s${reset}\n" "$@" ;}
+
+#-----------------------------------------------------------------------------------------                   
+# Functions
+#-----------------------------------------------------------------------------------------                   
+function usage {
+    h1 "Syntax"
+    cat << EOF
+set-version.sh [OPTIONS]
+Options are:
+-v | --version xxx : Mandatory. Set the version number to something explicitly. 
+    Re-builds the release.yaml based on the contents of sub-projects.
+    For example '--version 0.29.0'
+EOF
+}
+
+component_version=""
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -v | --version )        shift
+                                export component_version=$1
+                                ;;
+        -h | --help )           usage
+                                exit
+                                ;;
+        * )                     error "Unexpected argument $1"
+                                usage
+                                exit 1
+    esac
+    shift
+done
+
+if [[ -z $component_version ]]; then 
+    error "Missing mandatory '--version' argument."
+    usage
+    exit 1
+fi
+
+function update_build_gradle_version {
+    source_file="${BASEDIR}/galasa-parent/galasa-boot/build.gradle"
+    h1 "Updating the version in $source_file "
+
+    temp_dir=$1
+
+    set -o pipefail
+
+    temp_file="$temp_dir/build.gradle"
+
+    info "Using temporary file $temp_file"
+    info "Updating file $source_file"
+
+    cat $source_file | sed "s/version[ ]*=[ ]*.*$/version = '$component_version'/1" > $temp_file
+    rc=$?; if [[ "${rc}" != "0" ]]; then error "Failed to set version into $source_file file."; exit 1; fi
+    cp $temp_file ${source_file}
+    rc=$?; if [[ "${rc}" != "0" ]]; then error "Failed to overwrite new version of $source_file file."; exit 1; fi
+
+    success "$source_file updated OK."
+}
+
+function update_release_yaml_version {
+    source_file="${BASEDIR}/release.yaml"
+    h1 "Updating the version in $source_file "
+
+    temp_dir=$1
+
+    is_line_supressed=false
+
+    set -o pipefail
+
+    temp_file="$temp_dir/release.yaml"
+
+    info "Using temporary file $temp_file"
+    info "Updating file $source_file"
+
+    #Read through the realease yaml and search for line matching the regex. Marks the line after the match and overwrites it.
+    while IFS= read -r line
+    do
+        regex="^  - artifact: galasa-boot[ ]*$"
```

Comment:
You could probably simplify this regex to only look for `galasa-boot` rather than the entire `artifact` line, so the pattern could be something like:
```
^.*galasa-boot[ ]*$"
```

Reply:
I think I did it this way to be more specific, in case it ever gets mentioned again